exonum-java-binding-common/src/main/java/com/exonum/binding/common/serialization/ProtobufSerializer.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.common.serialization;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.MessageLite;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+
+class ProtobufSerializer<MessageT extends MessageLite> implements Serializer<MessageT> {
+
+  private final MethodHandle messageParseFrom;
```

Comment:
Might be useful to add implementation details to javadoc. It's not that obvious why do you save this method handle for future invocations.

Reply:
What exactly is not obvious?

Reply:
I can rename ProtobufSerializer to ProtobufReflectiveSerializer + document which method this handle is for.

Reply:
Agree to rename it to `ProtobufReflectiveSerializer`